cmd/vicadmin/vicadm.go

```diff
@@ -680,8 +694,17 @@ func (s *server) index(res http.ResponseWriter, req *http.Request) {
 func main() {
 	flag.Parse()
 
+	// create the appropriate authenticator
+	switch config.authType {
+	case "none":
+		config.Authenticator = &auth.None{}
+	case "basic":
+		config.Authenticator = auth.NewBasicHTTP("", "")
+	}
+
 	s := &server{
-		addr: config.addr,
+		Authenticator: config.Authenticator,
```

Comment:
Looks like this is the only place config.Authenticator is referenced. This could probably be eliminated in lieu of server.Authenticator.